R/g.getmeta.R

```diff
@@ -607,6 +607,41 @@ g.getmeta = function(datafile, params_metrics = c(), params_rawdata = c(),
           }
           CWav[h,1] = max(c(CW[h,1],CW[h,2],CW[h,3])) #indicator of clipping
         }
+        # Revise edges of nonwear score
+        # they might be imputed in g.imputeTimegaps and not revised for nonwear
+        # browser()
```

Comment:
This entire block of looking for edges is now run for all data formats. I think it is safer to make this specific to only recordings where g.imputeTimegaps was run and where timegaps were identified. So, maybe add:

```
if (exists("remaining_epochs")) {
}
```

Reply:
I started with this, but then tried this on other formats and the nonwear was also improved when long nonwear periods were detected, we are only double-checking the edges of the nonwear windows and there relaxing the sdcriter from being met for 1 hour to 15 minutes (in defaults). I think it is reasonable since 15 minutes meeting the nonwear criteria right before or after a long nonwear period is something to be expected when the device is not worn. So I thought we could do this on all formats (time processing does not get very affected), are you ok with this?

Reply:
This will affect methodological consistency, and will change the non-wear detection algorithm. I think you approach will make the algorithm hard to explain: Why are we applying a rolling window calculation of the standard deviation and then later look again at the standard deviation of raw data. Why does the first time we calculate sd fail to detect the non-wear?

Would it not be simpler to change the algortihm as follows:

`if (windowsizes[3] > 3 * windowsizes[2]) {` label the windowsizes[2] epoch before and after a long non-wear period as nonwear. So, instead of looking at the raw data again we update the values of `NWav`. I think this would be easier to explain and seen as a relatively simple update the existing algorithm.

If we do that we need to update vignette with a statement that the non-wear detection algorithm has been enhanced.

Reply:
Actually, we can then not limit it to long non-wear period but simply apply it to all detect non-wear periods, because all of them reflect a windowsizes[3] of a very low standard deviation.

Reply:
... Ok, after a bit more reflection I agree that your approach is better,  but my concerns then are: 
- It may be logical to do the same for the start of the window, because I think same logic applies there. If a person stops wearing the accelerometer at let's say 12:05 then this will only become visible in the non-wear detection at the long epoch that starts at 12:15.
- We need to make sure this is clearly described in the vignette and NEWS.Rd. For example, including a description of how we define long nonwear.

Reply:
Sounds good. I am working parallely in 2 branches and I improved the gap index detection in [issue643_longGap](https://github.com/wadpac/GGIR/tree/issue643_longGap). I would like to see how these 2 changes work together, so I am going to merge the changes in [issue643_longGap](https://github.com/wadpac/GGIR/tree/issue643_longGap) into [issue645_nonwear_edges](https://github.com/wadpac/GGIR/tree/issue645_nonwear_edges) and keep working on this. Once we do that, we can close the other PR since everything will be commited from here.